src/Chapter4.hs

```diff
@@ -489,6 +495,26 @@ Implement the 'Applicative' instance for our 'List' type.
   type.
 -}
 
+instance Applicative List where
+    pure :: a -> List a
+    pure a = Cons a Empty
+
+    (<*>) :: List (a -> b) -> List a -> List b
+    Empty <*> _ = Empty
+    _ <*> Empty = Empty
+    fs <*> vs = concatL (fmap (\v -> fmap (\f -> f v ) fs ) vs)
+
+
+concatL :: List (List a) -> List a
+concatL Empty = Empty
+concatL (Cons x Empty) = x
+concatL (Cons x (Cons y y')) = addLists x (addLists y (concatL y'))
+
+addLists :: List a -> List a -> List a
+addLists l Empty = l
+addLists Empty l = l
+addLists (Cons l Empty) r = Cons l r
+addLists (Cons l (Cons l' l'')) r = Cons l (Cons l' (addLists l'' r))
```

Comment:
I see that you handle cases of singleton lists and lists of at least two elements in these. I think, your implementations can be simplified, if you only consider plain `Cons x xs` constructors. You already call functions recursively, and they handle `Empty` lists quite well, so no need in having extra cases :slightly_smiling_face:

Reply:
You are right! Thank you